Documentation/project-docs/cross-platform-cryptography.md

```diff
@@ -0,0 +1,241 @@
+Windows CAPI is used on Windows whenever `new RSACryptoServiceProvider()` is used, as well as the `GetRSAPrivateKey()` extension method when the private key is not able to be opened with CNG (which requires a hardware-backed key with no CNG-capable driver).
+
+Windows CAPI is capable of PKCS1 signature with a SHA-2 algorithm, but the individual RSA object may be loaded in a CSP which does not support it.
+
+#### Native Interop
+
+.NET Core exposes types to allow programs to interoperate with the system libraries upon which the .NET cryptography code is layered.
+The types involved do not translate between platforms, and should only be directly used when necessary.
+
+| Type | Windows | Linux | macOS |
+|------|---------|-------|-------|
+| RSACryptoServiceProvider | :white_check_mark: | :question: | :question: |
+| RSACng | :white_check_mark: | :x: | :x: |
+| RSAOpenSsl | :x: | :white_check_mark: | :question: |
+
+RSAOpenSsl on macOS works if OpenSSL is installed in the system and an appropriate libcrypto dylib can be found via dynamic library loading, otherwise exceptions will be raised.
+
+On non-Windows systems RSACryptoServiceProvider can be used for compatibility with existing programs, but a `PlatformNotSupportedException` will be thrown from any method which requires system interop, such as opening a named key.
+
+### ECDSA
+
+ECDSA key generation is performed by the system libraries, and is subject to size limitations and performance characteristics thereof.
+ECDSA key curves are defined by the system libraries, and are subject to the limitations thereof.
+
+| EC Curve | Windows 10 | Linux | macOS | Windows 7 - 8.1 |
+|----------|------------|-------|-------|-----------------|
+| NIST P-256 (secp256r1) | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| NIST P-384 (secp384r1) | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| NIST P-521 (secp521r1) | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| brainpool curves (as named curves) | :white_check_mark: | :question: | :x: | :x: |
+| other named curves | :question: | :question: | :x: | :x: |
+| explicit curves | :white_check_mark: | :white_check_mark: | :x: | :x: |
+| Export or import as explicit | :white_check_mark: | :white_check_mark: | :x: | :x: |
+
+Support for named curves was added to Windows CNG in Windows 10, and is not available in prior OSes, with the exception of the three curves which had special support in Windows 7.
+See [CNG Named Elliptic Curves](https://msdn.microsoft.com/en-us/library/windows/desktop/mt632245(v=vs.85).aspx) for the expected support.
+
+Not all Linux distributions have support for the same named curves.
```

Comment:
No, I don't know of one that exists.  When we first started doing the ECCurve/ECParameters feature we noticed that CentOS only did the NIST/secP curves by name, and Ubuntu was happy with quite a lot of stuff.  So there's a compile-time option that some distros seem to have elected to use.